frontend/include/chpl/resolution/scope-types.h

```diff
@@ -566,6 +553,12 @@ class BorrowedIdsWithName {
  */
 using DeclMap = std::unordered_map<UniqueString, OwnedIdsWithName>;
 
+bool lookupInDeclMap(const DeclMap& declared,
+                     UniqueString name,
+                     MatchingIdsWithName& result,
+                     IdAndFlags::Flags filterFlags,
+                     const IdAndFlags::FlagSet& excludeFlags);
```

Comment:
We have `using DeclMap = std::unordered_map<UniqueString, OwnedIdsWithName>;` and I can't add methods to `unordered_map`. It might be worthwhile to make `DeclMap` a class, but I don't think that's called for yet.